src/govuk/components/button/button.test.js

```diff
@@ -27,112 +47,240 @@ describe('/components/button', () => {
+          window.__SUBMIT_EVENTS++
+          // Don't refresh the page, which will destroy the context to test against.
+          event.preventDefault()
+        })
+      })
+    }
+
+    /**
+       * Gets the number of times the form was submitted
+       *
+       * @returns {Number}
+       */
+    function getClicksCount () {
+      return page.evaluate(() => window.__SUBMIT_EVENTS)
+    }
+
+    describe('not enabled', () => {
+      it('does not prevent multiple submissions', async () => {
+        await page.goto(baseUrl + '/components/button/preview', {
+          waitUntil: 'load'
         })
 
+        await trackClicks()
+
         await page.click('button')
         await page.click('button')
 
-        const submitCount = await page.evaluate(() => window.__SUBMIT_EVENTS)
+        const clicksCount = await getClicksCount()
 
-        expect(submitCount).toBe(1)
+        expect(clicksCount).toBe(2)
       })
+    })
+
+    describe('using data-attributes', () => {
+      it('prevents unintentional submissions when in a form', async () => {
+        await page.goto(
+          baseUrl + '/components/button/prevent-double-click/preview',
+          { waitUntil: 'load' }
+        )
+
+        await trackClicks()
+
+        await page.click('button')
+        await page.click('button')
+
+        const clicksCount = await getClicksCount()
+
+        expect(clicksCount).toBe(1)
+      })
+
       it('does not prevent intentional multiple clicks', async () => {
-        await page.goto(baseUrl + '/components/button/prevent-double-click/preview', { waitUntil: 'load' })
+        await page.goto(
+          baseUrl + '/components/button/prevent-double-click/preview',
+          { waitUntil: 'load' }
+        )
 
-        // Our examples don't have form wrappers so we need to overwrite it.
+        await trackClicks()
+
+        await page.click('button')
+        await page.click('button', { delay: 1000 })
+
+        const clicksCount = await getClicksCount()
+
+        expect(clicksCount).toBe(2)
+      })
+
+      it('does not prevent subsequent clicks on different buttons', async () => {
+        await page.goto(
+          baseUrl + '/components/button/prevent-double-click/preview',
+          { waitUntil: 'load' }
+        )
+
+        await trackClicks()
+
+        // Clone button to have two buttons on the page
         await page.evaluate(() => {
           const $button = document.querySelector('button')
-          const $form = document.createElement('form')
-          $button.parentNode.appendChild($form)
-          $button.parentNode.removeChild($button)
-          $form.appendChild($button)
-
-          window.__SUBMIT_EVENTS = 0
-          $form.addEventListener('submit', event => {
-            window.__SUBMIT_EVENTS++
-            // Don't refresh the page, which will destroy the context to test against.
-            event.preventDefault()
-          })
+          const $buttonPrime = $button.cloneNode(true)
```

Comment:
It took me a while to work out what this meant (and I'm still not 100% sure – guessing this is Is prime as in 'a derivative'?)

Wondering if `$secondButton` might be simpler and clearer?